signature/types.go

```diff
@@ -197,17 +198,20 @@ func (signerInfo *SignerInfo) ExtendedAttribute(key string) (Attribute, error) {
 	return Attribute{}, errors.New("key not in ExtendedAttributes")
 }
 
-// AuthenticSigningTime returns the authentic signing time
+// AuthenticSigningTime returns the authentic signing time under signing scheme
+// notary.x509.signingAuthority.
+// For signing scheme notary.x509, since it only supports authentic timestamp,
+// an error is returned.
+//
+// Reference: https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#signing-time--authentic-signing-time
 func (signerInfo *SignerInfo) AuthenticSigningTime() (time.Time, error) {
-	switch signerInfo.SignedAttributes.SigningScheme {
-	case SigningSchemeX509SigningAuthority:
-		return signerInfo.SignedAttributes.SigningTime, nil
-	case SigningSchemeX509:
-		if len(signerInfo.UnsignedAttributes.TimestampSignature) > 0 {
-			// TODO: Add TSA support for AutheticSigningTime
-			// https://github.com/notaryproject/notation-core-go/issues/38
-			return time.Time{}, errors.New("TSA checking has not been implemented")
+	switch signingScheme := signerInfo.SignedAttributes.SigningScheme; {
+	case signingScheme == SigningSchemeX509SigningAuthority:
+		if signerInfo.SignedAttributes.SigningTime.IsZero() {
+			return time.Time{}, fmt.Errorf("authentic signing time must be present under signing scheme %q", signingScheme)
 		}
+		return signerInfo.SignedAttributes.SigningTime, nil
```

Comment:
nit: code style: readability: consider
```suggestion
		signingTime := signerInfo.SignedAttributes.SigningTime {
		if signingTime.IsZero() {
			return time.Time{}, fmt.Errorf("authentic signing time must be present under signing scheme %q", signingScheme)
		}
		return signingTime, nil
```